### ISSUE TITLE
[amp-autocomplete] should fire DOM `change` event on select

### ISSUE THREAD

**Issue by Joyining** [ampproject/amphtml#26273]
## What's the issue?

amp-autocomplete doesn't seem to fire `input` event.
I tried to combine `amp-autocomplete` and `amp-script` to get current input value. But the autocomplete action didn't trigger the `input` event, therefore I can only get the user original input (eg. user input 'o' and it's autocompleted to 'orange', but I only get 'o').  

## How do we reproduce the issue?

Here is the sample code:
I didn't use `<input>` to summit because I need two buttons which direct to different urls. Therefore I use two `<a>` and set the `href`.

```
<form action="XXXX" id="searchProductPc" method="GET" target="_top">
 <amp-autocomplete filter="substring">
  <amp-script script="dynamic-href">
    <div>
        <input class="search-bar" type="text" name="p" id="pcSearchInput">
        <a id="searchProductLink" href="">Search Product</a>
        <a id="searchStoreLink"  href="">Search Store</a>
    </div>  
  </amp-script> 
  <script type="application/json">
    {
      "items": ["apple", "orange", "banana"]
    }
  </script>
 </amp-autocomplete>
</form>

<script id="dynamic-href" type="text/plain" target="amp-script">
 const searchProductLink = document.getElementById('searchProductLink');
 const searchStoreLink = document.getElementById('searchStoreLink');
 const pcSearchInput = document.getElementById('pcSearchInput');

 pcSearchInput.addEventListener('input', (e) => {
  let inputvalue = pcSearchInput.value;
  console.log('input value: '+inputvalue);
  searchProductLink.setAttribute('href', 'https://www.example.com/product?p='+inputvalue);
  searchStoreLink.setAttribute('href', 'https://www.example.com/store?p='+inputvalue);
 });
</script>
```

## Which AMP version is affected?

amp-autocomplete-0.1

**Comment by caroqliu** [comment]
Thanks for reporting this issue! I was able to reproduce the issue. This should be resolved by firing a `change` event on the nested input element upon selection of an item in the `amp-autocomplete` suggestion menu. It is also worth considering setting a flag from the selection handler in the extension binary so the event listener can consume the native `change` event on the blur from the selection action, and then firing one programmatically from the selection handler with the updated value.

This is an excellent GFI, so I'll be tagging as such and happy to work with anyone who would like to contribute.

**Comment by mdmower** [comment]
@caroqliu This issue is bit more complex than emitting a new `change` event. Borrowing @Joyining's sample and adding a `change` listener...
```
 pcSearchInput.addEventListener('change', (e) => {
  let inputvalue = pcSearchInput.value;
  console.log('input value: '+inputvalue);
  searchProductLink.setAttribute('href', 'https://www.example.com/product?p='+inputvalue);
  searchStoreLink.setAttribute('href', 'https://www.example.com/store?p='+inputvalue);
 });
```
... you'll find that the `change` callback is triggered on mouse selection. Unfortunately, `mutateElement()` is called before input values are updated. Work-in-progress PR #26826 is started and feedback would be appreciated. @samouri - You've been looking at amp-script/amp-bind timing a lot lately, if you have any comments about ensuring amp-autocomplete orders events correctly (set input value, call mutateElement, fire other custom events), they'd be welcome!

A minimal example AMP document is included in PR #26826 for testing.
